### PR TITLE
Add basic functionality

### PR DESCRIPTION
**Pull request by gabriel-duque**
I just pushed some commits to start the project.

I basically added a skeleton we can build on.

In the source code I:

* added some logging, warning and error handling functions
* wrote a function to read assets (our words) in from different files
* parsed command line options so we can specify a file for the grammar and a file for assets
* added empty functions to do the rest

Everything seems to be fine for now!

This is a draft pull request so we shouldn't set it as mergeable untill it works well.

### CODE REVIEW

**Comment by gabriel-duque** [comment]
CHANGELOG:

* python classes now
* `-v` flag to turn on verbose
* initial test suite

**Comment by bandang0** [comment]
Hi dude,
Thanks for kicking off our project!

I just have a couple of questions:

__Technical questions:__
* On `aamg/generator.py`: we have this line:
> self.grammar: typing.Dict[str, str] = dict()

Should it be `typing.Dict[str, typing.List[str]]`?

* On `tests/lexer_test.py`: are there some `\n` missing in `file_content`?

__General questions:__
* On `ASSETS` and `GRAMMAR`: I think the end user will not use these particular grammars and asset files, and we may use others to illustrate the functioning of the project. So what do you say we have `examples/grammars` and `examples/assets` folders with these grammar and assets files (and others!)? Eventually, we could also put grammars and assets for tests in there.

* On `README.md`: I like the CHANGELOG you put in the thread up there, what do you say we have a CHANGELOG entry in the README, or a separate `CHANGES.md` file that we update every PR with significant improvements?

* On `setup.py` and `tests/`: first of all this mock interface is awesome, and I'm not used to testing (bad gollum!) but I'd like to get used to testing in this manner. Unfortunately, when you run `python3 setup.py test` (with Setuptools 45.2.0 and Python 3.8.1), you get:

> running test
> WARNING: Testing via this command is deprecated and will be removed in a future version.
> Users looking for a generic test entry point independent of test runner are encouraged to use
> tox.

As discussed in [this issue of setuptools](https://github.com/pypa/setuptools/issues/1684), the test command will be deprecated in further versions. Should we find some other handle for the tests (now or later)?

__Apart from the technical questions, I'm ready to rebase & merge this and address the general questions in later changes.__

File: aamg/generator.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    def parse_grammar(self) -> None:
+        '''Parse grammar file'''
+        if self.args.verbose:
+            log.info(f'loading grammar file: {self.args.grammar}')
+        try:
+            with open(self.args.grammar, 'r') as f:
+                lines = (line.strip().split('=') for line in f.readlines())
+        except OSError as e:
+            log.die(f'{self.args.grammar}: {e.strerror}')
+        self.grammar: typing.Dict[str, str] = dict()
```
Comment (bandang0) [inline aamg/generator.py]: Should this be typing.Dict[str, typing.List[str]] ?

File: tests/lexer_test.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        '''This is just the first thing I tried so it stayed here'''
+
+        expected_dict: typing.Dict[str, typing.List[str]] = dict()
+        expected_dict['model'] =   ['\'The\'', 'subject', '(', 'adv', '|', 
+                'adv', '\'and\'', 'adv', ')', 'verb', '(', '\'a\'', '|',
+                '\'theonly\'', ')', 'object']
+        expected_dict['subject'] = ['"The"', '\'sdf\''] 
+
+        file_content: str = str()
+        file_content += "model='The' subject ( adv | adv 'and' adv )  verb "
```
Comment (bandang0) [inline tests/lexer_test.py:74]: Is this lacking a \n at the end?

File: tests/lexer_test.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        file_content += "( 'a' | 'theonly' ) object\n"
+        file_content += "model='The' subject ( animal | adv ) verb \"ffds\"\n"
+        file_content += "subject=\"The\" 'sdf'\n"
+        file_content += "\n"
+        file_content += "### asdf\n"
+        file_content += "\n"
+        file_content += "# ignored\n"
+        file_content += "model='The' subject [ ( adv | adv 'and' adv ) ] verb "
```
Comment (bandang0) [inline tests/lexer_test.py:82]: idem

**Review by bandang0 — COMMENTED** [review]
As indicated in the thread of the PR, apart from the technical questions, I'm ready to rebase & merge this and solve the general questions later if you like @zuh0 .

File: aamg/generator.py (hunk 1 above)
Comment (gabriel-duque) [reply aamg/generator.py]: yup ! gonna fix this.

File: tests/lexer_test.py (hunk 2 above)
Comment (gabriel-duque) [reply tests/lexer_test.py:74]: nope I just didn't want to have a line over 80 columns so I split it

File: tests/lexer_test.py (hunk 3 above)
Comment (gabriel-duque) [reply tests/lexer_test.py:82]: same!

**Comment by gabriel-duque** [comment]
I just applied your fixes.

**Review by bandang0 — APPROVED** [review]
Alright here we go!

**Pull request merged**